datafusion/functions-window/src/nth_value.rs

```diff
@@ -360,9 +360,10 @@ impl PartitionEvaluator for NthValueEvaluator {
                     })
                     .unwrap_or_default();
                 if valid_indices.is_empty() {
-                    return ScalarValue::try_from(arr.data_type());
+                    None
```

Comment:
took me a while to figure out what's happening here; maybe instead check slice.nulls() explicitly and return None if it doesn't exist (i.e. there's no nulls)?

also the pre-existing short-circuiting might be useful in cases where the nulls() index exists but valid_indices() truly is empty (i.e. the array is completely nulls)?

Reply:
I filed https://github.com/apache/datafusion/issues/14063 to track the idea.  Thank you 🙏